allennlp/data/dataset_readers/snli.py

```diff
@@ -71,6 +71,10 @@ def text_to_instance(self,  # type: ignore
         hypothesis_tokens = self._tokenizer.tokenize(hypothesis)
         fields['premise'] = TextField(premise_tokens, self._token_indexers)
         fields['hypothesis'] = TextField(hypothesis_tokens, self._token_indexers)
+        fields['metadata'] = MetadataField({
```

Comment:
Instead of adding a `MetadataField` that the model doesn't really need, I think this would be cleaner if you just modified the [`Predictor`](https://github.com/allenai/allennlp/blob/master/allennlp/service/predictors/decomposable_attention.py).  The `Predictor` can tokenize the text and pass the tokens on as output to the demo.  That would look something like this: https://github.com/allenai/allennlp/blob/f246da737791ebe4fdbc6eca3119762eefa0c55b/allennlp/service/predictors/sentence_tagger.py#L27-L39

You'd have to grab the `_tokenizer` from `self._dataset_reader` in the `Predictor` (with `# pylint: disable=protected-access`) and use that to tokenize the text, instead of instantiating a tokenizer in `__init__`.  You'd also need to modify this `DatasetReader`'s `text_to_instance` method to optionally take pre-tokenized text, which would look something like how `passage_tokens` is treated here: https://github.com/allenai/allennlp/blob/f246da737791ebe4fdbc6eca3119762eefa0c55b/allennlp/data/dataset_readers/reading_comprehension/squad.py#L74-L83